Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OandaBrokerage"/> class.
+        /// </summary>
+        /// <param name="orderProvider">The order provider.</param>
+        /// <param name="holdingsProvider">The holdings provider.</param>
+        /// <param name="accountId">The account identifier.</param>
+        public OandaBrokerage(IOrderProvider orderProvider, IHoldingsProvider holdingsProvider, int accountId)
+            : base("Oanda Brokerage")
+        {
+            _orderProvider = orderProvider;
+            _holdingsProvider = holdingsProvider;
+            AccountId = accountId;
+            _cachedOpenOrdersByOandaOrderId = new ConcurrentDictionary<long, DataType.Order>();
+            InstrumentSecurityTypeMap =  new Dictionary<string, SecurityType>(); 
+        }
+
+        /// <summary>
+        /// Initializes the instrument security type map.
+        /// </summary>
+        public void InitializeInstrumentSecurityTypeMap()
+        {
+            InstrumentSecurityTypeMap = MapInstrumentToSecurityType(GetInstrumentsAsync());
+        }
+
+        /// <summary>
+        /// Returns true if we're currently connected to the broker
+        /// </summary>
+        public override bool IsConnected
+        {
+            get { return _issuedAt + _lifeSpan > DateTime.Now; }
+        }
+
+        /// <summary>
+        /// Gets all open orders on the account. 
+        /// NOTE: The order objects returned do not have QC order IDs.
+        /// </summary>
+        /// <returns>The open orders returned from Oanda</returns>
+        public override List<Order> GetOpenOrders()
+        {
+            var oandaOrders = GetOrderList();
+
+            var orderList = oandaOrders.Select(ConvertOrder).ToList();
+            foreach (var openOrder in orderList)
+            {
+                _cachedOpenOrdersByOandaOrderId[openOrder.Id] = oandaOrders.FirstOrDefault(oo => oo.id == openOrder.Id);
+            }
+            return orderList;
+        }
+
+        /// <summary>
+        /// Gets all holdings for the account
+        /// </summary>
+        /// <returns>The current holdings from the account</returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+            var holdings = GetPositions().Select(ConvertHolding).ToList();
+            return holdings;
+        }
+
+        /// <summary>
+        /// Converts the Oanda position into a QuantConnect holding.
+        /// </summary>
+        /// <param name="position">The position.</param>
+        /// <returns></returns>
+        protected Holding ConvertHolding(Position position)
+        {
+            return new Holding
+            {
+                Symbol = position.instrument,
+                Type = InstrumentSecurityTypeMap[position.instrument],
+                AveragePrice = (decimal)position.avgPrice,
+                ConversionRate = 1.0m,
+                CurrencySymbol = "$",
```

Comment:
In Oanda you can select the currency the account is represented in. It's usually in the following:

Australian Dollar (AUD)
Canadian Dollar (CAD)
Swiss Franc (CHF)
Euro (EUR)
British Pound (GBP)
Hong Kong Dollar (HKD)
Japanese Yen (JPY)
Singapore Dollar (SGD)
US Dollar (USD)

In QC is currently everything represented as USD?